content/en/blog/2023/dlb-connection-balance/index.md

```diff
@@ -0,0 +1,305 @@
+---
+title: "Using Accelerated Offload Connection Load Balance in Istio"
+description: "Accelerate connection balance using DLB connection balance configuration in Istio gateways."
+publishdate: 2023-06-08
+attribution: "Loong Dai (Intel)"
+keywords: [Istio, DLB, gateways]
+---
+
+## What is connection load balance?
+
+Load balancing is a core networking solution used to distribute traffic across multiple servers in a server farm. Load balancers improve application availability and responsiveness and prevent server overload. Each load balancer sits between client devices and backend servers, receiving and then distributing incoming requests to any available server capable of fulfilling them.
+
+For a common web server, it usually has multiple workers (processors or threads). If many clients connect to a single worker, this worker becomes busy and brings big tail latency while other workers run in the free state, affecting the performance of web server. Connection load balance is the solution for this situation, which is also known as connection balance.
+
+## What does Istio do for connection load balance?
+
+Istio uses Envoy as the data plane.
+
+Envoy provides a connection load balance implementation called Exact connection balance. As its name says, a lock is held during balancing so that connection counts are nearly exactly balanced between workers. It is "nearly" exact in the sense that a connection might close in parallel thus making the counts incorrect, but this should be rectified on the next accept. This balancer sacrifices accept throughput for accuracy and should be used when there are a small number of connections that rarely cycle, e.g., service mesh gRPC egress.
```

Comment:
```suggestion
Envoy provides a connection load balancing implementation called Exact connection balance. As its name says, a lock is held during balancing so that connection counts are nearly exactly balanced between workers. It is "nearly" exact in the sense that a connection might close in parallel thus making the counts incorrect, but this should be rectified on the next accept. This balancer sacrifices accept throughput for accuracy and should be used when there are a small number of connections that rarely cycle, e.g., service mesh gRPC egress.
```